File: apps/server/src/memory/tokenEstimate.ts
import { APPROX_TOKEN_PER_CHAR } from './constants.js';

export const estimateTextTokens = (text: string): number => {
  if (!text.trim()) {
    return 0;
  }

  return Math.max(1, Math.ceil(text.length * APPROX_TOKEN_PER_CHAR));
};

File: apps/server_py/app/memory/token_estimate.py
from app.memory.constants import APPROX_TOKEN_PER_CHAR


def estimate_text_tokens(text: str) -> int:
    if not text.strip():
        return 0
    return max(1, int((len(text) * APPROX_TOKEN_PER_CHAR) + 0.999))
